(TO) Repository-centric docs (#26402)
DataDog/documentation: content/en/tests/branches.md

---
title: Branches
description: Gain insight into test performance in your repository branches.
aliases:
- /tests/search/
algolia:
   rank: 70
   tags: ['flaky test', 'flaky tests', 'test regression', 'test regressions', 'test service', 'test services']
further_reading:
- link: "/continuous_integration/explorer"
  tag: "Documentation"
  text: "Search and filter test runs"
- link: "/continuous_integration/guides/flaky_test_management"
  tag: "Documentation"
  text: "Learn how to manage flaky tests"
- link: "/tests/repositories"
  tag: "Documentation"
  text: "Gain insight into test performance in your repositories"
---

## Overview

> The [Branches][1] view of the Tests page lists all branches from all {{< tooltip glossary="test service" >}}s that have reported test results. This tab is useful for seeing the status of tests run on your code branches and troubleshooting test failures.

Reviewer: ```suggestion
The [Branches][1] view of the Repositories page lists all branches from all repositories and {{< tooltip glossary="test service" >}}s that have reported test results. This tab is useful for seeing the status of tests run on your code branches and troubleshooting test failures.
```

To filter the list down to only branches that you have committed to, toggle **My Branches** and enter the email address associated with your GitHub account. You can enter multiple email addresses. Edit your address later by clicking **Edit Authors**.

## Test results

> For each branch, you can see the test service, the number of failed, passed, and skipped tests, test regressions, total test time, when the commit was last updated, and the avatar of the author of the commit.

Reviewer: ```suggestion
For each branch, you can see the repository, test service, the number of failed, passed, and skipped tests, test regressions, total test time, when the commit was last updated, and the avatar of the author of the commit.
```

> For each branch, you can see the test service, the number of failed, passed, and skipped tests, test regressions, total test time, when the commit was last updated, and the avatar of the author of the commit.

Reviewer: ```suggestion
For each branch, you can see the repository, test service, the number of failed, passed, and skipped tests, new flaky tests, test regressions, total test time, when the commit was last updated, and the avatar of the author of the commit.
```

> For each branch, you can see the test service, the number of failed, passed, and skipped tests, test regressions, total test time, when the commit was last updated, and the avatar of the author of the commit.

Reviewer: ```suggestion
For each branch, you can see the repositories test service(s), the number of failed, passed, and skipped tests, test regressions, total test time, when the commit was last updated, and the avatar of the author of the commit.
```

> Click on a branch to explore the test details page, which includes information about the branch's latest commits, flaky tests, test performance, common error types, and all test runs.

Reviewer: ```suggestion
Click on a branch to explore the commit details page, which includes information about the failed tests, new flaky tests, test regressions, test performance, related pipeline executions, and all test runs against that commit.
```
